source/client/HandManagement.py

```diff
@@ -0,0 +1,114 @@
+import pygame
+import client.Button as Btn
+import client.UIConstants as UIC
+from client.UICardWrapper import UICardWrapper
+
+"""This file contains methods used in displaying, selecting and sorting the hand"""
+
+
+def wrapHand(self, updated_hand, wrapped_hand):
+    """Associate each card in updated_hand with a UICardWrapper
+
+    if change is that new cards were added, then want to preserve location and status for cards that were
+    already in hand.
+    If cards were played, then the wrapped cards that are removed should to those cards with status = 2
+    (prepared cards).
+    Note that discards are removed immediately after controller confirms discard legal.
+    """
+    card_xy = (10, UIC.Table_Hand_Border + 40)
+    # sort cards so that if prepared cards were played, those are the instances of the cards that are removed.
+    old_wrapped_hand = sorted(wrapped_hand, key=lambda x: x.status)
+    updated_wrapped_hand = []
+    if not updated_hand == []:
+        for card in updated_hand:
+            newcard = True
+            for already_wrapped in old_wrapped_hand:
+                if newcard and card == already_wrapped.card:
+                    card_wrapped = already_wrapped
+                    card_xy = (max(card_xy[0], card_wrapped.img_clickable.x), card_xy[1])
+                    old_wrapped_hand.remove(already_wrapped)
+                    newcard = False
+            if newcard:
+                card_xy = (card_xy[0] + self.hand_scaling[1], card_xy[1])
+                card_wrapped = UICardWrapper(card, card_xy, self.hand_scaling[0])
+            updated_wrapped_hand.append(card_wrapped)
+        maxX = UIC.Disp_Width - (self.hand_scaling[1] / 2)
+        if (card_xy[0] > maxX):
+            scalingfactor = maxX / card_xy[0]
+            self.hand_scaling = (scalingfactor * self.hand_scaling[0], scalingfactor * self.hand_scaling[1])
+            updated_wrapped_hand = rescaleCards(self, updated_wrapped_hand, self.hand_scaling[0])
+            self.refresh_flag = True
+        if (self.hand_scaling[0] != UIC.scale and len(updated_wrapped_hand) <= self.deal_size):
+            self.hand_scaling = (UIC.scale, UIC.Card_Spacing)
+            updated_wrapped_hand = rescaleCards(self, updated_wrapped_hand, self.hand_scaling[0])
+        return updated_wrapped_hand
+
+def clearPreparedCardsGui(self):
```

Comment:
The Gui in this name is confusing, it makes me think its rendering a gui, not clearing UI Only statuses. It could clear both UI and controller, since you give it self. Also, I think you can pass it hand_info (which I think is the wrapped card list?) on its own without passing display if you keep it as clearing wrapped card status only.

Reply:
Renamed this method.  
As written, this method double checks that prepared cards are all successfully cleared before it will change status, but HandManagement doesn't talk to controller directly.

Reply:
Renamed this method.  Also changed so only passing hand_info.